src/operations/edit/editAll.ts

```diff
@@ -40,3 +42,26 @@ export function editAll<R, P>(ctx: HandlerContext,
     return doWithAllRepos<EditResult, P>(ctx, credentials, edit, parameters,
         repoFinder, repoFilter, repoLoader);
 }
+
+/**
+ * Edit the given repo with the given editor function, which depends only on the project
+ * @param credentials credentials
+ * @param {ProjectEditor} editor
+ * @param editInfo: EditMode determines how the edits should be applied.
+ * @param {RepoLoader} repoLoader (optional, useful in testing)
+ * @return {Promise<EditResult>}
+ */
+export function editOne(credentials: ProjectOperationCredentials,
+                        editor: (p: Project) => Promise<Project>,
+                        editInfo: EditMode,
+                        singleRepository: RepoRef,
+                        repoLoader: RepoLoader = defaultRepoLoader(credentials)): Promise<EditResult> {
+    const officialProjectEditor: ProjectEditor = (project, noContext, noParams) =>
+        Promise.resolve(editor(project)).then(successfulEdit, err => failedEdit(project, err));
+
+    const singleRepoFinder: RepoFinder = () => Promise.resolve([singleRepository]);
+
+    return editAll(null, credentials, officialProjectEditor, editInfo, {},
```

Comment:
Why do you not take a context to `editOne`? That breaks the editor contract. The fact that many editors don't use the context is beside the point.